packages/records/lib/models/records.ts

```diff
@@ -227,6 +240,21 @@ export async function upsert({
                         return false;
                     }
                 });
+
+                const delta_count = chunkSummary.addedKeys.length - (chunkSummary.deletedKeys?.length ?? 0);
+                await trx
+                    .from(RECORD_COUNTS_TABLE)
+                    .insert({
+                        connection_id: connectionId,
+                        model,
+                        environment_id: environmentId,
+                        object_count: Math.max(0, delta_count)
```

Comment:
Doing max here since the first time this runs, if the sync job happens to be deleting more than it adds, we'll end up with a negative count.

Reply:
is this possible? First run can't deleteKeys (AFAIK)
edit: maybe on the second generation and `markNonCurrentGenerationRecordsAsDeleted` kicks in? @TBonnin

Reply:
I was thinking if this is deployed to existing syncs, there will be no `record_counts` entry for the sync, and so this update will be starting from an incorrect count (hence why I think we need the temporary task to get the initial correct count for everybody first).

But for this specifically, I guess it doesn't make much sense to arbitrarily limit it to zero if the reported count can be inaccurate anyway

Reply:
Ah sorry - never got any of my notifications properly set up so missed this comment.

I'll talk about it during the presentation, but my plan was to set up a temporary "admin endpoint" that only we can hit (would need to understand details of how to set something like that up here), that would enqueue a job for every `(environment_id, connection_id)` pair. This job would get the full count for each model on that pair and update the `record_counts` table with that info. These jobs could be spread out over some amount of time (maybe an hour? maybe more) to avoid locking up database resources/locking the tables themselves for an extended period of time.